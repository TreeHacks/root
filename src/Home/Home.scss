@import "../Global.scss";

.nav {
  background-color: $background-color;
  height: 80px;
  width: 100vw;
  padding: 0 2vw;
  border-bottom: 3px solid $background-color-light-border;

  display: flex;
  flex-flow: row nowrap;
  justify-content: center;
  align-items: center;

  a {
    color: $text-color !important;
    margin: 0 $navbar-margin;
    padding: $navbar-padding 0;
    font-size: $navbar-font-size;
    border-bottom: 2px solid $background-color;
    transition: 0.5s;

    &:hover, &.active {
      border-bottom: 2px solid $primary-color;
      text-decoration: none;
    }
  }

  .header-logo {
    flex: 1;
    text-align: left;
    margin-left: -10px;

    a {
      border-bottom: 0px;
      margin: 0;
      font-size: 36px;
    }

    .logo-text-tree {
      font-family: $base-font-family;
    }

    .logo-text-hacks {
      font-family: 'Avenir Heavy';
    }
  }

  .logoutText {
    flex: 1;
    text-align: right;
    cursor: pointer;
    display: inline-block;
    padding: $navbar-padding/2 0;
  }
}


@media only screen and (max-width: 1024px) {
  .nav {
    flex-flow: column;
    height: auto;
    padding: 10px 0;
  }
} 