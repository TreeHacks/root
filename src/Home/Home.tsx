import React from "react";
import { connect } from 'react-redux';
import { logo } from "../constants";
import { logout } from "../store/auth/actions";

import { IHomeProps } from "./types";
import "./Home.scss";
import { NavLink } from "react-router-dom";

const mapStateToProps = state => ({
  ...state.home,
  auth: state.auth
});

const mapDispatchToProps = (dispatch, ownProps) => ({
  logout: () => dispatch(logout()),
});

export class Home extends React.Component<IHomeProps, {}> {
  componentDidMount() {
  }

  render() {
    return <div className="nav">
      <div className="header-logo">
        <NavLink to="/">
          <img src={logo} height="40px" />
          <span className="logo-text-tree">tree</span>
          <span className="logo-text-hacks">hacks</span>
        </NavLink>
      </div>
      <div className="treehacks-navbar-links">
        {this.props.auth.applicant && <NavLink to="/" isActive={(_, loc) => loc.pathname === "/"}>dashboard</NavLink>}
        {this.props.auth.applicant && <NavLink to="/application_info">application</NavLink>}
        {this.props.auth.applicant && <NavLink to="/transportation">travel</NavLink>}
        {this.props.auth.admin &&
          <NavLink to="/admin">admin</NavLink>
        }
        {this.props.auth.reviewer &&
          <NavLink to="/review">review</NavLink>
        }
        {this.props.auth.sponsor &&
          <NavLink to="/sponsors">sponsors</NavLink>
        }
        {this.props.auth.judge &&
          <NavLink to="/judge">judge</NavLink>
        }
      </div>
      <div className="logoutText">
        <a onClick={() => this.props.logout()}>log out</a>
      </div>
    </div>
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Home);
