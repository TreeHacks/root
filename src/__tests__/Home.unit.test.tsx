import React from "react";
import { shallow } from 'enzyme';
import { IHomeProps } from "../Home/types";
import { Home } from "../Home/Home";
import {NavLink} from "react-router-dom";

const props: IHomeProps = {
    profile: null,
    schemas: null,
    page: null,
    formData: null,
    formName: null,
    userEdited: null,
    auth: {
        loggedIn: false,
        user: null,
        userId: "test",
        admin: false,
        reviewer: false,
        sponsor: false,
        judge: false,
        applicant: false,
    },
    getUserProfile: () => null,
    logout: () => null,
};

describe.only('Home screen tests', () => {
    it('home screen applicant', () => {
        const wrapper = shallow(
            <Home {...{ ...props, auth: { ...props.auth, applicant: true } }} />
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(NavLink).length).toEqual(4);
        // expect(wrapper.find(NavLink).at(1).text()).toEqual("dashboard");
        // expect(wrapper.find(NavLink).at(2).text()).toEqual("application");
        // expect(wrapper.find(NavLink).at(3).text()).toEqual("travel");
    });
    it('home screen admin', () => {
        const wrapper = shallow(
            <Home {...{ ...props, auth: { ...props.auth, applicant: true, admin: true } }} />
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(NavLink).length).toEqual(5);
        // expect(wrapper.find(NavLink).at(4).text()).toEqual("admin");
    });
    it('home screen reviewer', () => {
        const wrapper = shallow(
            <Home {...{ ...props, auth: { ...props.auth, applicant: true, reviewer: true } }} />
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(NavLink).length).toEqual(5);
        // expect(wrapper.find(NavLink).at(4).text()).toEqual("reviewer");
    });
    it('home screen sponsor', () => {
        const wrapper = shallow(
            <Home {...{ ...props, auth: { ...props.auth, sponsor: true } }} />
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(NavLink).length).toEqual(2);
        // expect(wrapper.find(NavLink).at(1).text()).toEqual("sponsor");
    });
    it('home screen judge', () => {
        const wrapper = shallow(
            <Home {...{ ...props, auth: { ...props.auth, judge: true } }} />
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(NavLink).length).toEqual(2);
        // expect(wrapper.find(NavLink).at(1).text()).toEqual("judge");
    });
});