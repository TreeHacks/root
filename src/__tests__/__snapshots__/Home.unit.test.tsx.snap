// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen tests home screen admin 1`] = `
<div
  className="nav"
>
  <div
    className="header-logo"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/"
    >
      <img
        height="40px"
        src={[Function]}
      />
      <span
        className="logo-text-tree"
      >
        tree
      </span>
      <span
        className="logo-text-hacks"
      >
        hacks
      </span>
    </NavLink>
  </div>
  <div
    className="treehacks-navbar-links"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      isActive={[Function]}
      to="/"
    >
      dashboard
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/application_info"
    >
      application
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/transportation"
    >
      travel
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/admin"
    >
      admin
    </NavLink>
    <a
      className="logout"
      onClick={[Function]}
    >
      log out
    </a>
  </div>
</div>
`;

exports[`Home screen tests home screen applicant 1`] = `
<div
  className="nav"
>
  <div
    className="header-logo"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/"
    >
      <img
        height="40px"
        src={[Function]}
      />
      <span
        className="logo-text-tree"
      >
        tree
      </span>
      <span
        className="logo-text-hacks"
      >
        hacks
      </span>
    </NavLink>
  </div>
  <div
    className="treehacks-navbar-links"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      isActive={[Function]}
      to="/"
    >
      dashboard
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/application_info"
    >
      application
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/transportation"
    >
      travel
    </NavLink>
    <a
      className="logout"
      onClick={[Function]}
    >
      log out
    </a>
  </div>
</div>
`;

exports[`Home screen tests home screen judge 1`] = `
<div
  className="nav"
>
  <div
    className="header-logo"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/"
    >
      <img
        height="40px"
        src={[Function]}
      />
      <span
        className="logo-text-tree"
      >
        tree
      </span>
      <span
        className="logo-text-hacks"
      >
        hacks
      </span>
    </NavLink>
  </div>
  <div
    className="treehacks-navbar-links"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/judge"
    >
      judge
    </NavLink>
    <a
      className="logout"
      onClick={[Function]}
    >
      log out
    </a>
  </div>
</div>
`;

exports[`Home screen tests home screen reviewer 1`] = `
<div
  className="nav"
>
  <div
    className="header-logo"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/"
    >
      <img
        height="40px"
        src={[Function]}
      />
      <span
        className="logo-text-tree"
      >
        tree
      </span>
      <span
        className="logo-text-hacks"
      >
        hacks
      </span>
    </NavLink>
  </div>
  <div
    className="treehacks-navbar-links"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      isActive={[Function]}
      to="/"
    >
      dashboard
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/application_info"
    >
      application
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/transportation"
    >
      travel
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/review"
    >
      review
    </NavLink>
    <a
      className="logout"
      onClick={[Function]}
    >
      log out
    </a>
  </div>
</div>
`;

exports[`Home screen tests home screen sponsor 1`] = `
<div
  className="nav"
>
  <div
    className="header-logo"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/"
    >
      <img
        height="40px"
        src={[Function]}
      />
      <span
        className="logo-text-tree"
      >
        tree
      </span>
      <span
        className="logo-text-hacks"
      >
        hacks
      </span>
    </NavLink>
  </div>
  <div
    className="treehacks-navbar-links"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      to="/sponsors"
    >
      sponsors
    </NavLink>
    <a
      className="logout"
      onClick={[Function]}
    >
      log out
    </a>
  </div>
</div>
`;
